src/MessageDispatchingEventDispatcher.php

```diff
@@ -15,19 +15,20 @@ public function __construct(MessageDispatcher $dispatcher, ?MessageDecorator $de
         $this->decorator = $decorator ?: new DefaultHeadersDecorator();
     }
 
-    public function dispatch(object ...$events): void
+    public function dispatch(iterable|object $events): void
     {
-        $this->dispatchWithHeaders([], ...$events);
+        $this->dispatchWithHeaders([], $events);
     }
 
-    public function dispatchWithHeaders(array $headers, object ...$events): void
+    public function dispatchWithHeaders(array $headers, iterable|object $events): void
     {
+        $events = is_iterable($events) ? $events : [$events];
         $messages = [];
 
         foreach ($events as $event) {
             $messages[] = $this->decorator->decorate(new Message($event, $headers));
         }
```

Comment:
To keep leveraging generators, maybe this could pass messages as a generator instead?

```suggestion
    $messages = (function() use ($events, $headers): Generator {
        foreach ($events as $event) {
            yield $this->decorator->decorate(new Message($event, $headers));
        }
    })();
```
```